packages/grafana-data/src/transformations/transformers/seriesToColumns.test.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  ArrayVector,
+  DataTransformerConfig,
+  DataTransformerID,
+  FieldType,
+  toDataFrame,
+  transformDataFrame,
+} from '@grafana/data';
+import { SeriesToColumnsOptions } from './seriesToColumns';
+
+describe('SeriesToColumns Transformer', () => {
+  const everySecondSeries = toDataFrame({
+    name: 'even',
+    fields: [
+      { name: 'time', type: FieldType.time, values: [3000, 4000, 5000, 6000] },
+      { name: 'temperature', type: FieldType.number, values: [10.3, 10.4, 10.5, 10.6] },
+    ],
+  });
+
+  const everyOtherSecondSeries = toDataFrame({
+    name: 'odd',
+    fields: [
+      { name: 'time', type: FieldType.time, values: [1000, 3000, 5000, 7000] },
+      { name: 'temperature', type: FieldType.number, values: [11.1, 11.3, 11.5, 11.7] },
+    ],
+  });
+
+  it('joins by time field', () => {
+    const cfg: DataTransformerConfig<SeriesToColumnsOptions> = {
+      id: DataTransformerID.seriesToColumns,
+      options: {
+        byField: 'time',
+      },
+    };
+
+    const filtered = transformDataFrame([cfg], [everySecondSeries, everyOtherSecondSeries])[0];
+    expect(filtered.fields).toEqual([
+      {
+        name: 'time',
+        type: FieldType.time,
+        values: new ArrayVector([1000, 3000, 4000, 5000, 6000, 7000]),
+        config: {},
+        labels: { origin: 'even|odd' },
```

Comment:
nit:  maybe 'even,odd' rather than | -- it seems more like an "and" rather than an "or"

Reply:
Yeah, I had it that way at first, but then I went for the `|` for no particular reason. I'll change.